rest.go

```diff
@@ -0,0 +1,297 @@
+package gosysl
+
+import (
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/anz-bank/gosysl/pb"
+)
+
+type route struct {
+	methods         map[string]string
+	middleware      string
+	keys            []string
+	queryParams     map[string][]string
+	postPayloadType string
+	putPayloadType  string
+}
+
+type routes struct {
+	paths   []string
+	content map[string]*route
+}
+
+var validHTTPMethods = map[string]interface{}{
+	"Get":    struct{}{},
```

Comment:
If we're being picky, HTTP methods should be all uppercase: GET, PUT, …

Reply:
I'm a champion in pickiness... done.